objects/src/accounts/storage/slot.rs

```diff
@@ -1,175 +1,48 @@
-use alloc::string::{String, ToString};
-
-use miden_crypto::merkle::SimpleSmt;
 use vm_core::{
     utils::{Deserializable, Serializable},
     Word, EMPTY_WORD,
 };
-use vm_processor::DeserializationError;
-
-use super::{map::EMPTY_STORAGE_MAP_ROOT, Felt, STORAGE_TREE_DEPTH};
-
-// CONSTANTS
-// ================================================================================================
-
-const MAX_VALUE_ARITY: u8 = u8::MAX - 1;
-
-const MIN_ARRAY_DEPTH: u8 = 2;
-const MAX_ARRAY_DEPTH: u8 = 64;
 
-const DEFAULT_SLOT_TYPE: StorageSlotType = StorageSlotType::Value { value_arity: 0 };
+use super::{map::EMPTY_STORAGE_MAP_ROOT, StorageMap};
 
-// STORAGE SLOT TYPE
+// STORAGE SLOT
 // ================================================================================================
 
-/// An object that represents the type of a storage slot.
+/// An enum that represents the type of a storage slot and it's value.
 #[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
-#[derive(Debug, Copy, Clone, PartialEq, Eq)]
-pub enum StorageSlotType {
-    /// Represents a slot that contains a value with the specified arity.
-    Value { value_arity: u8 },
-    /// Represents a slot that contains a commitment to a map with values with the specified arity.
-    Map { value_arity: u8 },
-    /// Represents a slot that contains a commitment to an array with capacity 2^depth with values
-    /// with the specified arity.
-    Array { depth: u8, value_arity: u8 },
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum StorageSlot {
+    Value(Word),
+    Map(StorageMap),
 }
```

Comment:
I think we may still need to keep `StorageSlotType` enum. For example, it would be used in `AccountStorage::get_slot_type()` or `AccountStorage::layout()`.

We could also have methods like `StorageSlotType::as_element()` and `StorageSlotType::as_word()` to simplify conversion of storage into a vector of elements.